backend/user/middleware.go

```diff
@@ -135,78 +137,114 @@ func NewUserMiddleware(queries *sqlc.Queries, memberLoader *loaders.Loader[int,
-			roles = append(roles, RoleBCCMember)
-		}
+			roles = append(roles, RoleRegistered)
+			if ctx.GetBool(auth0.CtxIsBCCMember) {
+				roles = append(roles, RoleBCCMember)
+			}
 
-		u := &common.User{
-			PersonID:  pid,
-			Roles:     roles,
-			Anonymous: false,
-			ActiveBCC: ctx.GetBool(auth0.CtxIsBCCMember),
-			AgeGroup:  "unknown",
-		}
+			u := &common.User{
+				PersonID:  pid,
+				Roles:     roles,
+				Anonymous: false,
+				ActiveBCC: ctx.GetBool(auth0.CtxIsBCCMember),
+				AgeGroup:  "unknown",
+			}
 
-		member, err := memberLoader.Get(ctx, int(intID))
-		if err != nil {
-			log.L.Error().Err(err).Msg("Failed to retrieve user")
-			span.AddEvent("User failed to load")
-			userCache.Set(userID, u, cache.WithExpiration(1*time.Minute))
-			ctx.Set(CtxUser, u)
-			return
-		}
+			member, err := ls.MemberLoader.Get(ctx, int(intID))
+			if err != nil {
+				log.L.Info().Err(err).Msg("Failed to retrieve user from members.")
+				span.AddEvent("User failed to load from members")
 
-		email := member.Email
+				dbUser, err := ls.UserLoader.Get(ctx, pid)
+				if err != nil {
+					log.L.Error().Err(err).Msg("Failed to retrieve user from database.")
+				}
+				if dbUser != nil {
+					u = dbUser
+				}
 
-		if email == "" {
-			// Explicit values make it easier to see that it was intended when debugging
-			email = "<MISSING>"
-		}
+				userCache.Set(userID, u, cache.WithExpiration(1*time.Minute))
+				ctx.Set(CtxUser, u)
+				return u, nil
+			}
 
-		userRoles, err := GetRolesForEmail(reqCtx, queries, email)
-		if err != nil {
-			err = merry.Wrap(err)
-			log.L.Warn().Err(err).Str("email", email).Msg("Unable to get roles")
-		} else {
-			roles = append(roles, userRoles...)
-		}
+			email := member.Email
 
-		u.Roles = roles
-		u.Email = email
-		u.DisplayName = member.DisplayName
+			if email == "" {
+				// Explicit values make it easier to see that it was intended when debugging
+				email = "<MISSING>"
+			}
 
-		// Set AgeGroup and avoid passing identifying information through the application
-		if member.BirthDate != "" {
-			birthDate, err := time.Parse("2006-01-02", member.BirthDate)
+			userRoles, err := GetRolesForEmail(reqCtx, queries, email)
 			if err != nil {
-				log.L.Error().Err(err).Msg("Error parsing birthday of user")
+				err = merry.Wrap(err)
+				log.L.Warn().Err(err).Str("email", email).Msg("Unable to get roles")
 			} else {
-				u.Age = time.Now().Year() - birthDate.Year()
-				ageGroupMin := 0
-				for minAge, group := range AgeGroups {
-					// Note: Maps are not iterated in a sorted order, so we have to find the lowed applicable
-					if u.Age >= minAge && minAge > ageGroupMin {
-						u.AgeGroup = group
-						ageGroupMin = minAge
+				roles = append(roles, userRoles...)
+			}
+
+			u.Roles = roles
+			u.Email = email
+			u.DisplayName = member.DisplayName
+
+			// Set AgeGroup and avoid passing identifying information through the application
+			if member.BirthDate != "" {
```

Comment:
Are you aware that members provides a field called `qge` so we could drop the whole parsing part?